src/stable/test/index.ts

```diff
@@ -39,50 +39,60 @@ export function runTests(tests: Test): void {
     (shouldRunTests === true ? describe : describe.skip)(
         `correctness tests`,
         () => {
-            (shouldCheckMemoryStateAfterTests === true ? please : please.skip)(
-                'snapshot the canister memory size and heap allocation for all canisters before correctness tests',
-                async () => {
+            if (shouldCheckMemoryStateAfterTests === true) {
+                beforeAll(async () => {
+                    console.info(
+                        'Preparing: snapshot the canister memory size and heap allocation for all canisters before correctness tests'
+                    );
                     const snapshot = await takeMemorySnapshot();
 
                     startingMemoryState.correctness.heapAllocations =
                         snapshot.heapAllocations;
                     startingMemoryState.correctness.memorySizes =
                         snapshot.memorySizes;
-                }
-            );
+                });
+            }
 
             tests();
 
-            (shouldCheckGlobalStateAfterTests === true ? it : it.skip)(
-                'checks that all internal global state variables for all canisters are empty',
-                runGlobalStateChecks
-            );
-
-            (shouldCheckMemoryStateAfterTests === true ? it : it.skip)(
-                'checks the canister memory size and heap allocation for all canisters after correctness tests',
-                async () => {
-                    await checkMemoryChanges(
-                        startingMemoryState.correctness.heapAllocations,
-                        startingMemoryState.correctness.memorySizes
-                    );
-                }
-            );
+            if (
+                shouldCheckGlobalStateAfterTests === true ||
+                shouldCheckMemoryStateAfterTests === true ||
+                shouldRunTypeChecks === true
+            ) {
+                afterAll(async () => {
+                    if (shouldCheckGlobalStateAfterTests === true) {
+                        console.info(
+                            'Testing: checks that all internal global state variables for all canisters are empty'
```

Comment:
Make sure to check your work by running a couple tests that have nested describes. Then look at the jest output, and ensure that the snapshot ran before all nested describes and tests, and that the global state and memory checks happened after.